src/app/catalog/catalog.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams, HttpHeaders } from '@angular/common/http';
+
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+import { Book } from './catalog';
+import { LocalstorageService } from '../services/localstorage.service';
+import { Router } from '@angular/router';
+
+@Injectable()
+export class CatalogService {
+  catalogUrl = 'https://js-band-store-api.glitch.me/books';
+
+  constructor(
+    private http: HttpClient,
+    private localStorageService: LocalstorageService,
+    private router: Router
+  ) {}
+
+  getCatalog(): Observable<Book[]> {
+    const options = {
+      headers: new HttpHeaders({
+        Authorization: `Bearer ${
+          this.localStorageService.getLocalData()?.token
```

Comment:
you should not add header here -> it should be a part of Auth module - > need to implement AuthInterceptor which will append this header for each and every GET request at least